### ISSUE TITLE
Test Plan: Hyperbolic-Conic Geodesic Mapping and Prime Clustering

### ISSUE THREAD

**Issue by zfifteen** [zfifteen/unified-framework#619]
⸻

✅ Test Plan: Hyperbolic-Conic Geodesic Mapping and Prime Clustering

🎯 Hypothesis

Reframing θ′(n, k) geodesic mappings in terms of conic eccentricity (e > 1)—analogous to slicing a cone hyperbolically—enhances prime clustering beyond classical k-based methods (e.g. k = 0.3), with statistically significant gains in density, reduced entropy, and stronger alignment to zeta zero spacings.

⸻

🔬 Scientific Objective

Establish whether hyperbolic parameterizations of geodesic mappings (θ′(n,e)) result in:
	1.	Increased prime density enhancement
	2.	Lower entropy in θ′ distributions
	3.	Higher statistical correlation with zeta zero spacings

Versus baseline:
	•	θ′(n, k = 0.3) (standard)
	•	θ′(n, k = kappa_star ≈ 0.04449) (Z5D-calibrated)

⸻

🧪 Test Protocol

1. Geodesic Mapping Definitions

Test 3 geodesic mappings:

A. Classical:

\theta′_k(n) = \phi \cdot \left( \frac{n \bmod \phi}{\phi} \right)^k

B. Hyperbolic-Conic Proposed:

\theta′_e(n) = \phi \cdot \left( \frac{n}{\phi} \right)^{1/e} \bmod 1 \quad\text{for } e > 1

C. Control:
	•	Uniform distribution
	•	Randomized composite set

⸻

2. Parameter Sweep

Parameter	Range	Step	Notes
e (eccentricity)	[1.01, 1.05, 1.1, 1.25, 1.5, 2.0]	discrete	conic curvature grows with e
N (number of primes)	[10⁴, 10⁵, 10⁶, 10⁷]	exponential	test convergence
Bins	50	fixed	modulo-1 fractional binning

⸻

3. Metrics to Measure

📊 A. Prime Density Enhancement
	•	Max bin % over uniform expectation (2%)
	•	Report absolute % enhancement over uniform

📉 B. Entropy
	•	Shannon entropy of binned θ′ values
	•	Compare ΔEntropy vs baseline

📈 C. Zeta Correlation
	•	Pearson r between θ′(n) values and normalized ζ-zero spacings (first 1K or 10K zeros)
	•	Compare r to baseline (expected r ≈ 0.93 for best θ′)

🧪 D. Bootstrap Confidence Intervals
	•	For all metrics, run 1,000 resamples and report 95% CI

🧮 E. KS Divergence (Prime vs Composite θ′)
	•	KS statistic: primes vs matched composite set (control)
	•	Significance: p < 0.01

⸻

📍 Implementation Skeleton

Data
	•	Use validated primes from all.txt or sympy.primerange
	•	Use composite controls matched in size and distribution

Code Snippet

import numpy as np
from sympy import primerange
from scipy.stats import entropy, ks_2samp, pearsonr
from sklearn.utils import resample
from mpmath import mp

mp.dps = 50
phi = (1 + mp.sqrt(5)) / 2

def theta_prime_e(n, e):
    return float(phi * (n / phi) ** (1/e) % 1)

def compute_metrics(primes, e, bins=50):
    thetas = [theta_prime_e(p, e) for p in primes]
    hist, _ = np.histogram(thetas, bins=bins, range=(0, 1), density=True)
    max_bin = np.max(hist)
    expected = 1.0
    enhancement = (max_bin - expected) / expected * 100
    shannon = entropy(hist)
    return enhancement, shannon, hist

# Example usage
primes = list(primerange(2, 10**6))
enhancement, shannon, hist = compute_metrics(primes, e=1.25)
print(f"Enhancement: {enhancement:.2f}%, Entropy: {shannon:.4f}")

⸻

📈 Validation Criteria

Metric	Success Threshold
Density Enhancement	> baseline (3.56% @ k=0.3), ideally ≥ 5%
Entropy	Lower than k=0.3 baseline
Zeta r	r > 0.93 with p < 10⁻⁶
KS Test	p < 0.01 for prime vs. composite θ′(n)
Bootstrap CI	Non-overlapping with control in ≥ 3 metrics

⸻

🧪 Bootstrap Framework (Example)

def bootstrap_ci(data, stat_fn, n=1000, alpha=0.05):
    stats = [stat_fn(resample(data)) for _ in range(n)]
    lo = np.percentile(stats, 100 * alpha / 2)
    hi = np.percentile(stats, 100 * (1 - alpha / 2))
    return lo, hi

⸻

✅ Success Criteria
	•	✅ Hypothesis Supported if:
	•	≥ 3 of 4 metrics show statistically significant improvement over θ′(k=0.3)
	•	Results consistent across N ∈ [10⁴, 10⁷]
	•	Bootstrap CI for enhancement does not overlap with baseline
	•	❌ Hypothesis Falsified if:
	•	No metrics exceed baseline across all e
	•	Bootstrap CI overlaps in all cases
	•	Zeta r decreases (e.g., r < 0.85)

⸻

🔄 Extension: Fractal-Conic Mapping

Pending results, test further variants:
	•	\theta′(n, e, f) = \phi \cdot \left( \frac{n \bmod \phi^f}{\phi^f} \right)^{1/e}
	•	Introduce fractional modularity or scaling

⸻

Would you like this packaged into:
	•	✅ A downloadable Python test script
	•	✅ An open GitHub Issue using your standard format
	•	✅ A LaTeX/Markdown scientific preprint draft for sharing
